[Bug]: Using --convert with a causal LM is silently accepted but crashes at engine startup

### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
Collecting environment information...
==============================
        System Info
==============================
OS                           : Ubuntu 22.04.3 LTS (x86_64)
GCC version                  : (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version                : Could not collect
CMake version                : version 3.22.1
Libc version                 : glibc-2.35

==============================
       PyTorch Info
==============================
PyTorch version              : 2.9.1+cu128
Is debug build               : False
CUDA used to build PyTorch   : 12.8
ROCM used to build PyTorch   : N/A
XPU used to build PyTorch    : N/A

==============================
      Python Environment
==============================
Python version               : 3.12.12 | packaged by Anaconda, Inc. | (main, Oct 21 2025, 20:16:04) [GCC 11.2.0] (64-bit runtime)
Python platform              : Linux-6.8.0-110-generic-x86_64-with-glibc2.35
    
==============================
       CUDA / GPU Info
==============================
Is CUDA available            : True
CUDA runtime version         : Could not collect
CUDA_MODULE_LOADING set to   : 
GPU models and configuration : GPU 0: NVIDIA H100 80GB HBM3
Nvidia driver version        : 570.124.06
cuDNN version                : Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.9.0
HIP runtime version          : N/A
MIOpen runtime version       : N/A
Is XNNPACK available         : True

==============================
          CPU Info
==============================
Architecture:                            x86_64
CPU op-mode(s):                          32-bit, 64-bit
Address sizes:                           52 bits physical, 57 bits virtual
Byte Order:                              Little Endian
CPU(s):                                  128
On-line CPU(s) list:                     0-127
Vendor ID:                               GenuineIntel
Model name:                              Intel(R) Xeon(R) Gold 6430
CPU family:                              6
Model:                                   143
Thread(s) per core:                      2
Core(s) per socket:                      32
Socket(s):                               2
Stepping:                                8
CPU max MHz:                             3400.0000
CPU min MHz:                             800.0000
BogoMIPS:                                4200.00
Flags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cat_l2 cdp_l3 intel_ppin cdp_l2 ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect user_shstk avx_vnni avx512_bf16 wbnoinvd dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req vnmi avx512vbmi umip pku ospke waitpkg avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid bus_lock_detect cldemote movdiri movdir64b enqcmd fsrm md_clear serialize tsxldtrk pconfig arch_lbr ibt amx_bf16 avx512_fp16 amx_tile amx_int8 flush_l1d arch_capabilities ibpb_exit_to_user
Virtualization:                          VT-x
L1d cache:                               3 MiB (64 instances)
L1i cache:                               2 MiB (64 instances)
L2 cache:                                128 MiB (64 instances)
L3 cache:                                120 MiB (2 instances)
NUMA node(s):                            2
NUMA node0 CPU(s):                       0-31,64-95
NUMA node1 CPU(s):                       32-63,96-127
Vulnerability Gather data sampling:      Not affected
Vulnerability Indirect target selection: Not affected
Vulnerability Itlb multihit:             Not affected
Vulnerability L1tf:                      Not affected
Vulnerability Mds:                       Not affected
Vulnerability Meltdown:                  Not affected
Vulnerability Mmio stale data:           Not affected
Vulnerability Reg file data sampling:    Not affected
Vulnerability Retbleed:                  Not affected
Vulnerability Spec rstack overflow:      Not affected
Vulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:                Mitigation; Enhanced / Automatic IBRS; IBPB conditional; PBRSB-eIBRS SW sequence; BHI BHI_DIS_S
Vulnerability Srbds:                     Not affected
Vulnerability Tsa:                       Not affected
Vulnerability Tsx async abort:           Not affected
Vulnerability Vmscape:                   Mitigation; IBPB before exit to userspace

==============================
Versions of relevant libraries
==============================
[pip3] flashinfer-python==0.5.3
[pip3] numpy==2.2.6
[pip3] nvidia-cublas-cu12==12.8.4.1
[pip3] nvidia-cuda-cupti-cu12==12.8.90
[pip3] nvidia-cuda-nvrtc-cu12==12.8.93
[pip3] nvidia-cuda-runtime-cu12==12.8.90
[pip3] nvidia-cudnn-cu12==9.10.2.21
[pip3] nvidia-cudnn-frontend==1.18.0
[pip3] nvidia-cufft-cu12==11.3.3.83
[pip3] nvidia-cufile-cu12==1.13.1.3
[pip3] nvidia-curand-cu12==10.3.9.90
[pip3] nvidia-cusolver-cu12==11.7.3.90
[pip3] nvidia-cusparse-cu12==12.5.8.93
[pip3] nvidia-cusparselt-cu12==0.7.1
[pip3] nvidia-cutlass-dsl==4.3.5
[pip3] nvidia-ml-py==13.590.48
[pip3] nvidia-nccl-cu12==2.27.5
[pip3] nvidia-nvjitlink-cu12==12.8.93
[pip3] nvidia-nvshmem-cu12==3.3.20
[pip3] nvidia-nvtx-cu12==12.8.90
[pip3] pyzmq==27.1.0
[pip3] torch==2.9.1+cu128
[pip3] torchaudio==2.9.1+cu128
[pip3] torchvision==0.24.1+cu128
[pip3] transformers==4.57.6
[pip3] triton==3.5.1
[conda] flashinfer-python         0.5.3                    pypi_0    pypi
[conda] numpy                     2.2.6                    pypi_0    pypi
[conda] nvidia-cublas-cu12        12.8.4.1                 pypi_0    pypi
[conda] nvidia-cuda-cupti-cu12    12.8.90                  pypi_0    pypi
[conda] nvidia-cuda-nvrtc-cu12    12.8.93                  pypi_0    pypi
[conda] nvidia-cuda-runtime-cu12  12.8.90                  pypi_0    pypi
[conda] nvidia-cudnn-cu12         9.10.2.21                pypi_0    pypi
[conda] nvidia-cudnn-frontend     1.18.0                   pypi_0    pypi
[conda] nvidia-cufft-cu12         11.3.3.83                pypi_0    pypi
[conda] nvidia-cufile-cu12        1.13.1.3                 pypi_0    pypi
[conda] nvidia-curand-cu12        10.3.9.90                pypi_0    pypi
[conda] nvidia-cusolver-cu12      11.7.3.90                pypi_0    pypi
[conda] nvidia-cusparse-cu12      12.5.8.93                pypi_0    pypi
[conda] nvidia-cusparselt-cu12    0.7.1                    pypi_0    pypi
[conda] nvidia-cutlass-dsl        4.3.5                    pypi_0    pypi
[conda] nvidia-ml-py              13.590.48                pypi_0    pypi
[conda] nvidia-nccl-cu12          2.27.5                   pypi_0    pypi
[conda] nvidia-nvjitlink-cu12     12.8.93                  pypi_0    pypi
[conda] nvidia-nvshmem-cu12       3.3.20                   pypi_0    pypi
[conda] nvidia-nvtx-cu12          12.8.90                  pypi_0    pypi
[conda] pyzmq                     27.1.0                   pypi_0    pypi
[conda] torch                     2.9.1+cu128              pypi_0    pypi
[conda] torchaudio                2.9.1+cu128              pypi_0    pypi
[conda] torchvision               0.24.1+cu128             pypi_0    pypi
[conda] transformers              4.57.6                   pypi_0    pypi
[conda] triton                    3.5.1                    pypi_0    pypi

==============================
         vLLM Info
==============================
ROCM Version                 : Could not collect
vLLM Version                 : 0.14.1
vLLM Build Flags:
  CUDA Archs: Not Set; ROCm: Disabled; XPU: Disabled
GPU Topology:
        GPU0    CPU Affinity    NUMA Affinity   GPU NUMA ID
GPU0     X      32-63,96-127    1               N/A

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

==============================
     Environment Variables
==============================
PYTORCH_NVML_BASED_CUDA_CHECK=1
TORCHINDUCTOR_COMPILE_THREADS=1
```

</details>


### 🐛 Describe the bug

When serving a causal LM (e.g. `Meta-Llama-3.1-8B-Instruct`) with `--convert classify`, `--convert embed`, or `--convert reward`, vLLM's CLI and `validate_parsed_serve_args` both accept the configuration without complaint, but the server then crashes during engine initialization with an internal `AssertionError: pooler_config is not None` in `adapters.py`.

The configuration is presented to users as a supported adaptation path (vLLM's own docs and config code reference `--convert` for pooling tasks), so this crash is unexpected and misleading.

Notable: `--convert reward` is explicitly rewritten to `embed` internally with a deprecation warning, yet still crashes for the same reason — confirming this is a validation gap, not a model-specific edge case.

## Steps to Reproduce

**Case 1: `--convert classify`**
```bash
vllm serve <path-to-Meta-Llama-3.1-8B-Instruct> \
  --host 127.0.0.1 \
  --port 19088 \
  --convert classify
```

**Case 2: `--convert embed`**
```bash
vllm serve <path-to-Meta-Llama-3.1-8B-Instruct> \
  --host 127.0.0.1 \
  --port 19089 \
  --convert embed
```

**Case 3: `--convert reward`**
```bash
vllm serve <path-to-Meta-Llama-3.1-8B-Instruct> \
  --host 127.0.0.1 \
  --port 19092 \
  --convert reward
```

All three commands are accepted by the CLI and pass argument validation, but the server exits before becoming healthy.

## Expected Behavior

One of the following should happen:

1. vLLM successfully starts with the requested `--convert` mode, or
2. vLLM rejects the configuration early with a clear, user-facing error explaining that the specified `--convert` mode is not supported for this model architecture.

In either case, the server should not proceed into deep engine and model initialization only to fail with an internal assertion.

## Actual Behavior

The server starts initialization but crashes in `vllm/model_executor/models/adapters.py` before becoming healthy. The crash trace ends with:

```
root@lizhiyuan-ubuntu-01:~# /root/anaconda3/envs/vllm/bin/vllm serve \
  /root/.cache/modelscope/hub/models/LLM-Research/Meta-Llama-3___1-8B-Instruct \
  --host 127.0.0.1 \
  --port 19092 \
  --convert reward
(APIServer pid=806751) INFO 05-13 10:26:00 [api_server.py:1272] vLLM API server version 0.14.1
(APIServer pid=806751) INFO 05-13 10:26:00 [utils.py:263] non-default args: {'model_tag': '/root/.cache/modelscope/hub/models/LLM-Research/Meta-Llama-3___1-8B-Instruct', 'host': '127.0.0.1', 'port': 19092, 'model': '/root/.cache/modelscope/hub/models/LLM-Research/Meta-Llama-3___1-8B-Instruct', 'convert': 'reward'}
(APIServer pid=806751) WARNING 05-13 10:26:00 [model.py:846] `--convert reward` is deprecated and will be removed in v0.15. Please use `--convert embed` instead.
(APIServer pid=806751) INFO 05-13 10:26:00 [model.py:530] Resolved architecture: LlamaForCausalLM
(APIServer pid=806751) INFO 05-13 10:26:00 [model.py:1545] Using max model len 131072
(APIServer pid=806751) INFO 05-13 10:26:01 [scheduler.py:229] Chunked prefill is enabled with max_num_batched_tokens=8192.
(APIServer pid=806751) INFO 05-13 10:26:01 [vllm.py:630] Asynchronous scheduling is enabled.
(APIServer pid=806751) INFO 05-13 10:26:01 [vllm.py:637] Disabling NCCL for DP synchronization when using async scheduling.
(EngineCore_DP0 pid=806992) INFO 05-13 10:26:11 [core.py:97] Initializing a V1 LLM engine (v0.14.1) with config: model='/root/.cache/modelscope/hub/models/LLM-Research/Meta-Llama-3___1-8B-Instruct', speculative_config=None, tokenizer='/root/.cache/modelscope/hub/models/LLM-Research/Meta-Llama-3___1-8B-Instruct', skip_tokenizer_init=False, tokenizer_mode=auto, revision=None, tokenizer_revision=None, trust_remote_code=False, dtype=torch.bfloat16, max_seq_len=131072, download_dir=None, load_format=auto, tensor_parallel_size=1, pipeline_parallel_size=1, data_parallel_size=1, disable_custom_all_reduce=False, quantization=None, enforce_eager=False, enable_return_routed_experts=False, kv_cache_dtype=auto, device_config=cuda, structured_outputs_config=StructuredOutputsConfig(backend='auto', disable_fallback=False, disable_any_whitespace=False, disable_additional_properties=False, reasoning_parser='', reasoning_parser_plugin='', enable_in_reasoning=False), observability_config=ObservabilityConfig(show_hidden_metrics_for_version=None, otlp_traces_endpoint=None, collect_detailed_traces=None, kv_cache_metrics=False, kv_cache_metrics_sample=0.01, cudagraph_metrics=False, enable_layerwise_nvtx_tracing=False, enable_mfu_metrics=False, enable_mm_processor_stats=False, enable_logging_iteration_details=False), seed=0, served_model_name=/root/.cache/modelscope/hub/models/LLM-Research/Meta-Llama-3___1-8B-Instruct, enable_prefix_caching=True, enable_chunked_prefill=True, pooler_config=None, compilation_config={'level': None, 'mode': <CompilationMode.VLLM_COMPILE: 3>, 'debug_dump_path': None, 'cache_dir': '', 'compile_cache_save_format': 'binary', 'backend': 'inductor', 'custom_ops': ['none'], 'splitting_ops': ['vllm::unified_attention', 'vllm::unified_attention_with_output', 'vllm::unified_mla_attention', 'vllm::unified_mla_attention_with_output', 'vllm::mamba_mixer2', 'vllm::mamba_mixer', 'vllm::short_conv', 'vllm::linear_attention', 'vllm::plamo2_mamba_mixer', 'vllm::gdn_attention_core', 'vllm::kda_attention', 'vllm::sparse_attn_indexer'], 'compile_mm_encoder': False, 'compile_sizes': [], 'compile_ranges_split_points': [8192], 'inductor_compile_config': {'enable_auto_functionalized_v2': False, 'combo_kernels': True, 'benchmark_combo_kernel': True}, 'inductor_passes': {}, 'cudagraph_mode': <CUDAGraphMode.FULL_AND_PIECEWISE: (2, 1)>, 'cudagraph_num_of_warmups': 1, 'cudagraph_capture_sizes': [1, 2, 4, 8, 16, 24, 32, 40, 48, 56, 64, 72, 80, 88, 96, 104, 112, 120, 128, 136, 144, 152, 160, 168, 176, 184, 192, 200, 208, 216, 224, 232, 240, 248, 256, 272, 288, 304, 320, 336, 352, 368, 384, 400, 416, 432, 448, 464, 480, 496, 512], 'cudagraph_copy_inputs': False, 'cudagraph_specialize_lora': True, 'use_inductor_graph_partition': False, 'pass_config': {'fuse_norm_quant': False, 'fuse_act_quant': False, 'fuse_attn_quant': False, 'eliminate_noops': True, 'enable_sp': False, 'fuse_gemm_comms': False, 'fuse_allreduce_rms': False}, 'max_cudagraph_capture_size': 512, 'dynamic_shapes_config': {'type': <DynamicShapesType.BACKED: 'backed'>, 'evaluate_guards': False, 'assume_32_bit_indexing': True}, 'local_cache_dir': None}
(EngineCore_DP0 pid=806992) INFO 05-13 10:26:11 [parallel_state.py:1214] world_size=1 rank=0 local_rank=0 distributed_init_method=tcp://172.17.0.4:39639 backend=nccl
(EngineCore_DP0 pid=806992) INFO 05-13 10:26:11 [parallel_state.py:1425] rank 0 in world size 1 is assigned as DP rank 0, PP rank 0, PCP rank 0, TP rank 0, EP rank N/A
(EngineCore_DP0 pid=806992) INFO 05-13 10:26:12 [gpu_model_runner.py:3808] Starting to load model /root/.cache/modelscope/hub/models/LLM-Research/Meta-Llama-3___1-8B-Instruct...
(EngineCore_DP0 pid=806992) INFO 05-13 10:26:12 [cuda.py:351] Using FLASH_ATTN attention backend out of potential backends: ('FLASH_ATTN', 'FLASHINFER', 'TRITON_ATTN', 'FLEX_ATTENTION')
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936] EngineCore failed to start.
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936] Traceback (most recent call last):
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/v1/engine/core.py", line 927, in run_engine_core
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]     engine_core = EngineCoreProc(*args, engine_index=dp_rank, **kwargs)
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/v1/engine/core.py", line 692, in __init__
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]     super().__init__(
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/v1/engine/core.py", line 106, in __init__
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]     self.model_executor = executor_class(vllm_config)
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/v1/executor/abstract.py", line 101, in __init__
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]     self._init_executor()
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/v1/executor/uniproc_executor.py", line 48, in _init_executor
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]     self.driver_worker.load_model()
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/v1/worker/gpu_worker.py", line 274, in load_model
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]     self.model_runner.load_model(eep_scale_up=eep_scale_up)
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/v1/worker/gpu_model_runner.py", line 3827, in load_model
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]     self.model = model_loader.load_model(
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]                  ^^^^^^^^^^^^^^^^^^^^^^^^
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/model_executor/model_loader/base_loader.py", line 50, in load_model
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]     model = initialize_model(
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]             ^^^^^^^^^^^^^^^^^
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/model_executor/model_loader/utils.py", line 48, in initialize_model
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]     return model_class(vllm_config=vllm_config, prefix=prefix)
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/model_executor/models/adapters.py", line 189, in __init__
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]     self._init_pooler(vllm_config, prefix=prefix)
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/model_executor/models/adapters.py", line 260, in _init_pooler
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]     assert pooler_config is not None
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936]            ^^^^^^^^^^^^^^^^^^^^^^^^^
(EngineCore_DP0 pid=806992) ERROR 05-13 10:26:13 [core.py:936] AssertionError
(EngineCore_DP0 pid=806992) Process EngineCore_DP0:
(EngineCore_DP0 pid=806992) Traceback (most recent call last):
(EngineCore_DP0 pid=806992)   File "/root/anaconda3/envs/vllm/lib/python3.12/multiprocessing/process.py", line 314, in _bootstrap
(EngineCore_DP0 pid=806992)     self.run()
(EngineCore_DP0 pid=806992)   File "/root/anaconda3/envs/vllm/lib/python3.12/multiprocessing/process.py", line 108, in run
(EngineCore_DP0 pid=806992)     self._target(*self._args, **self._kwargs)
(EngineCore_DP0 pid=806992)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/v1/engine/core.py", line 940, in run_engine_core
(EngineCore_DP0 pid=806992)     raise e
(EngineCore_DP0 pid=806992)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/v1/engine/core.py", line 927, in run_engine_core
(EngineCore_DP0 pid=806992)     engine_core = EngineCoreProc(*args, engine_index=dp_rank, **kwargs)
(EngineCore_DP0 pid=806992)                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(EngineCore_DP0 pid=806992)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/v1/engine/core.py", line 692, in __init__
(EngineCore_DP0 pid=806992)     super().__init__(
(EngineCore_DP0 pid=806992)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/v1/engine/core.py", line 106, in __init__
(EngineCore_DP0 pid=806992)     self.model_executor = executor_class(vllm_config)
(EngineCore_DP0 pid=806992)                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
(EngineCore_DP0 pid=806992)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/v1/executor/abstract.py", line 101, in __init__
(EngineCore_DP0 pid=806992)     self._init_executor()
(EngineCore_DP0 pid=806992)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/v1/executor/uniproc_executor.py", line 48, in _init_executor
(EngineCore_DP0 pid=806992)     self.driver_worker.load_model()
(EngineCore_DP0 pid=806992)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/v1/worker/gpu_worker.py", line 274, in load_model
(EngineCore_DP0 pid=806992)     self.model_runner.load_model(eep_scale_up=eep_scale_up)
(EngineCore_DP0 pid=806992)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/v1/worker/gpu_model_runner.py", line 3827, in load_model
(EngineCore_DP0 pid=806992)     self.model = model_loader.load_model(
(EngineCore_DP0 pid=806992)                  ^^^^^^^^^^^^^^^^^^^^^^^^
(EngineCore_DP0 pid=806992)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/model_executor/model_loader/base_loader.py", line 50, in load_model
(EngineCore_DP0 pid=806992)     model = initialize_model(
(EngineCore_DP0 pid=806992)             ^^^^^^^^^^^^^^^^^
(EngineCore_DP0 pid=806992)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/model_executor/model_loader/utils.py", line 48, in initialize_model
(EngineCore_DP0 pid=806992)     return model_class(vllm_config=vllm_config, prefix=prefix)
(EngineCore_DP0 pid=806992)            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(EngineCore_DP0 pid=806992)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/model_executor/models/adapters.py", line 189, in __init__
(EngineCore_DP0 pid=806992)     self._init_pooler(vllm_config, prefix=prefix)
(EngineCore_DP0 pid=806992)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/model_executor/models/adapters.py", line 260, in _init_pooler
(EngineCore_DP0 pid=806992)     assert pooler_config is not None
(EngineCore_DP0 pid=806992)            ^^^^^^^^^^^^^^^^^^^^^^^^^
(EngineCore_DP0 pid=806992) AssertionError
[rank0]:[W513 10:26:13.677509938 ProcessGroupNCCL.cpp:1524] Warning: WARNING: destroy_process_group() was not called before program exit, which can leak resources. For more info, please see https://pytorch.org/docs/stable/distributed.html#shutdown (function operator())
(APIServer pid=806751) Traceback (most recent call last):
(APIServer pid=806751)   File "/root/anaconda3/envs/vllm/bin/vllm", line 7, in <module>
(APIServer pid=806751)     sys.exit(main())
(APIServer pid=806751)              ^^^^^^
(APIServer pid=806751)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/entrypoints/cli/main.py", line 73, in main
(APIServer pid=806751)     args.dispatch_function(args)
(APIServer pid=806751)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/entrypoints/cli/serve.py", line 60, in cmd
(APIServer pid=806751)     uvloop.run(run_server(args))
(APIServer pid=806751)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/uvloop/__init__.py", line 96, in run
(APIServer pid=806751)     return __asyncio.run(
(APIServer pid=806751)            ^^^^^^^^^^^^^^
(APIServer pid=806751)   File "/root/anaconda3/envs/vllm/lib/python3.12/asyncio/runners.py", line 195, in run
(APIServer pid=806751)     return runner.run(main)
(APIServer pid=806751)            ^^^^^^^^^^^^^^^^
(APIServer pid=806751)   File "/root/anaconda3/envs/vllm/lib/python3.12/asyncio/runners.py", line 118, in run
(APIServer pid=806751)     return self._loop.run_until_complete(task)
(APIServer pid=806751)            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(APIServer pid=806751)   File "uvloop/loop.pyx", line 1518, in uvloop.loop.Loop.run_until_complete
(APIServer pid=806751)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/uvloop/__init__.py", line 48, in wrapper
(APIServer pid=806751)     return await main
(APIServer pid=806751)            ^^^^^^^^^^
(APIServer pid=806751)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/entrypoints/openai/api_server.py", line 1319, in run_server
(APIServer pid=806751)     await run_server_worker(listen_address, sock, args, **uvicorn_kwargs)
(APIServer pid=806751)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/entrypoints/openai/api_server.py", line 1338, in run_server_worker
(APIServer pid=806751)     async with build_async_engine_client(
(APIServer pid=806751)                ^^^^^^^^^^^^^^^^^^^^^^^^^^
(APIServer pid=806751)   File "/root/anaconda3/envs/vllm/lib/python3.12/contextlib.py", line 210, in __aenter__
(APIServer pid=806751)     return await anext(self.gen)
(APIServer pid=806751)            ^^^^^^^^^^^^^^^^^^^^^
(APIServer pid=806751)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/entrypoints/openai/api_server.py", line 173, in build_async_engine_client
(APIServer pid=806751)     async with build_async_engine_client_from_engine_args(
(APIServer pid=806751)                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(APIServer pid=806751)   File "/root/anaconda3/envs/vllm/lib/python3.12/contextlib.py", line 210, in __aenter__
(APIServer pid=806751)     return await anext(self.gen)
(APIServer pid=806751)            ^^^^^^^^^^^^^^^^^^^^^
(APIServer pid=806751)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/entrypoints/openai/api_server.py", line 214, in build_async_engine_client_from_engine_args
(APIServer pid=806751)     async_llm = AsyncLLM.from_vllm_config(
(APIServer pid=806751)                 ^^^^^^^^^^^^^^^^^^^^^^^^^^
(APIServer pid=806751)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/v1/engine/async_llm.py", line 205, in from_vllm_config
(APIServer pid=806751)     return cls(
(APIServer pid=806751)            ^^^^
(APIServer pid=806751)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/v1/engine/async_llm.py", line 132, in __init__
(APIServer pid=806751)     self.engine_core = EngineCoreClient.make_async_mp_client(
(APIServer pid=806751)                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(APIServer pid=806751)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/v1/engine/core_client.py", line 122, in make_async_mp_client
(APIServer pid=806751)     return AsyncMPClient(*client_args)
(APIServer pid=806751)            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
(APIServer pid=806751)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/v1/engine/core_client.py", line 824, in __init__
(APIServer pid=806751)     super().__init__(
(APIServer pid=806751)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/v1/engine/core_client.py", line 479, in __init__
(APIServer pid=806751)     with launch_core_engines(vllm_config, executor_class, log_stats) as (
(APIServer pid=806751)          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(APIServer pid=806751)   File "/root/anaconda3/envs/vllm/lib/python3.12/contextlib.py", line 144, in __exit__
(APIServer pid=806751)     next(self.gen)
(APIServer pid=806751)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/v1/engine/utils.py", line 921, in launch_core_engines
(APIServer pid=806751)     wait_for_engine_startup(
(APIServer pid=806751)   File "/root/anaconda3/envs/vllm/lib/python3.12/site-packages/vllm/v1/engine/utils.py", line 980, in wait_for_engine_startup
(APIServer pid=806751)     raise RuntimeError(
(APIServer pid=806751) RuntimeError: Engine core initialization failed. See root cause above. Failed core proc(s): {}
```

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

Hi — I'd like to take this. The fix looks like it belongs in the early validation path (`validate_parsed_serve_args` / the `--convert` handling) — when `--convert classify|embed|reward` is requested but the model architecture can't produce a `pooler_config`, raise a clear user-facing `ValueError` there instead of letting it crash with `AssertionError: pooler_config is not None` in `adapters.py`. I'd add the check plus a unit test covering the three convert modes against a causal-LM config.

Happy to defer if someone's already on it or if you'd prefer a different shape (e.g. making it actually work rather than reject) — wanted to confirm before opening a PR.

--- a/vllm/config/model.py
+++ b/vllm/config/model.py
@@ -546,7 +546,9 @@ class ModelConfig:
         is_generative_model = registry.is_text_generation_model(architectures, self)
         is_pooling_model = registry.is_pooling_model(architectures, self)
 
-        self.runner_type = self._get_runner_type(architectures, self.runner)
+        self.runner_type = self._get_runner_type(
+            architectures, self.runner, self.convert
+        )
         self.convert_type = self._get_convert_type(
             architectures, self.runner_type, self.convert
         )
@@ -884,11 +886,15 @@ class ModelConfig:
         self,
         architectures: list[str],
         runner: RunnerOption,
+        convert: ConvertOption,
     ) -> RunnerType:
         if runner != "auto":
             return runner
 
-        runner_type = self._get_default_runner_type(architectures)
+        if convert in {"auto", "none"}:
+            runner_type = self._get_default_runner_type(architectures)
+        else:
+            runner_type = "pooling"
 
         # Don't log the most common case
         if runner_type != "generate":

